sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQuery.scala

```diff
@@ -142,10 +143,19 @@ trait StreamingQuery {
   def processAllAvailable(): Unit
 
   /**
-   * Stops the execution of this query if it is running. This method blocks until the threads
-   * performing execution has stopped.
+   * Stops the execution of this query if it is running. This awaits the termination of the query
+   * execution threads or a timeout is hit.
+   *
+   * By default stop will block indefinitely. You can make it timeout (ms) by setting the
```

Comment:
You can configure timeout by configuration ...

Given the configuration takes timeConf, unit may not be necessary here.